### PR TITLE
DAOS-623 build: Warn and exit if scons_local is not present

### PR DESCRIPTION
**Pull request by jolivier23**
We don't support the "traditional build" anymore so change the
code so it warns if scons_local is not present or is not
initialized.

Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>

### CODE REVIEW

File: SConstruct [hunk 1]
```diff
@@ -19,6 +19,12 @@ DESIRED_FLAGS = ['-Wno-gnu-designator',
 PP_ONLY_FLAGS = ['-Wno-parentheses-equality', '-Wno-builtin-requires-header',
                  '-Wno-unused-function']
 
+def need_scons_local():
+    """Print scons_local warning and exit"""
+    print ('scons_local submodule is needed in order to do DAOS build')
+    print ('Use git submodule init; git submodule update')
```
Comment (brianjmurrell) [inline SConstruct]: Really just informational and I'm not sure if this operation as expressed in two separate `git submodule` commands will ever go away but I have observed:

```suggestion
    print ('Use git submodule update --init')
```

as an alternative.

Again, mostly just FYI.

But more to the point, is there any reason we wouldn't just do the `git submodule init; git submodule update` here for the user, or is the horse out of the barn at this point?

File: SConstruct [hunk 2]
```diff
@@ -238,13 +244,15 @@ def scons():
         exit(0)
 
     """Execute build"""
-    if os.path.exists('scons_local'):
+    if not os.path.exists('scons_local'):
+        need_scons_local()
+    else:
```
Comment (brianjmurrell) [inline SConstruct]: Isn't this covered by the `ImportError` below?

File: SConstruct (hunk 1 above)
Comment (jolivier23) [reply SConstruct]: I didn't know that was an option so thanks for pointing it out.   I don't think we should do it automatically for them.   If, for example, I'm doing local experiments on something (maybe another submodule), I might be annoyed if a build changed the state

File: SConstruct (hunk 2 above)
Comment (jolivier23) [reply SConstruct]: hmm, maybe it is.

**Comment by jolivier23** [comment]
Ok, fixed it...since travis build failed anyway

**Pull request merged**